### PR TITLE
Mask boolean default on

### LINKED ISSUE #2052: 🐛 FSL-BET `-m` flag (`mask_boolean`) should not be accessible to users
### Describe the bug

The `-m` [toggle](https://github.com/FCP-INDI/C-PAC/blob/v1.8.6/CPAC/resources/configs/pipeline_config_blank.yml#L349-L350) should always be On, but furthermore, there is no need for this option to be open as a switch to users. This step should always produce the binary brain mask instead of the already-skullstripped brain, due to C-PAC's dedicated [brain extraction block](https://github.com/FCP-INDI/C-PAC/blob/v1.8.6/CPAC/anat_preproc/anat_preproc.py#L1974-L2031).

Because the possible complication: using a config (such as the `fmriprep-options` preconfig which has `-m` set to Off for some reason) and then modifying it to use `FSL-BET` will lead to no mask being generated, and the skull-stripped brain being passed downstream as a mask.

### To reproduce

_No response_

### Preconfig

- [ ] default
- [ ] abcd-options
- [ ] anat-only
- [ ] blank
- [ ] ccs-options
- [ ] fmriprep-options
- [ ] fx-options
- [ ] monkey
- [ ] monkey-ABCD
- [ ] ndmg
- [ ] nhp-macaque
- [ ] preproc
- [ ] rbc-options
- [ ] rodent

### Custom pipeline configuration

_No response_

### Run command

_No response_

### Expected behavior

This should just need to be inaccessible as an option, with the `-m` flag hard-coded to On.

### Acceptance criteria

- [ ] Hard-code the `-m` flag in the `FSL BET` [block](https://github.com/FCP-INDI/C-PAC/blob/v1.8.6/CPAC/anat_preproc/anat_preproc.py#L618-L620).
- [ ] Make sure the `mask_boolean` keys in all of the configs are removed.

### Screenshots

_No response_

### C-PAC version

_No response_

### Container platform

_No response_

### Docker and/or Singularity version(s)

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by birajstha**
## Fixes

Fixes #2052 by @sgiavasis 

## Description

- Hard-coded the -m flag in the FSL BET [block](https://github.com/FCP-INDI/C-PAC/blob/v1.8.6/CPAC/anat_preproc/anat_preproc.py#L618-L620).
- The `mask_boolean` keys in all of the configs are removed.

## Tests

Run one of the `config` and check to see `working` directory to see if mask image is there inside the `anat_BET_skullstrip` folder

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository. 
- [x] My commit messages follow best practices.
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I updated the changelog.
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by shnizzedy — CHANGES_REQUESTED** [review]
This looks good!

- [ ] We should at least remove https://github.com/FCP-INDI/C-PAC/blob/cda064cf7b0bfbe8b837f75eca6d8572e6f6c6be/CPAC/pipeline/schema.py#L527 from the schema validator.

   Even better, once we update the [relevant docs](https://fcp-indi.github.io/docs/nightly/user/anat#configuring-fsl-bet-options) (I'd add a [`versionremoved` directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionremoved) above [this line](https://github.com/FCP-INDI/fcp-indi.github.io/blob/ba441d5f6082935b27157cf1e0ba339a9b54e299/docs/_sources/user/anat.rst?plain=1#L188) and move that line into the directive's contents with a little explanatory note lifted / adapted from #2052. In case you haven't worked on the docs yet, they're in a separate repo with [its own contributing guide](https://github.com/FCP-INDI/fcp-indi.github.io/blob/source/CONTRIBUTING.md) for its quirks). Once the docs are updated, we can add an exception handler like

   ```Python
           if (multiple_invalid.path == ['anatomical_preproc', 'brain_extraction', 'FSL-BET', 'mask_boolean']):
               raise Invalid("'mask_boolean' was removed as an option for FSL-BET in C-PAC"
                             f" v1.8.7. See {DOCS_URL_PREFIX}/user/anat#configuring-fsl-bet-options for more information.", path=multiple_invalid.path) from multiple_invalid
   ```

   before or after this if block https://github.com/FCP-INDI/C-PAC/blob/cda064cf7b0bfbe8b837f75eca6d8572e6f6c6be/CPAC/pipeline/schema.py#L1105-L1111 so if a user tries to provide that key, they'll get an error like

   ```
   Invalid: 'mask_boolean' was removed as an option for FSL-BET in C-PAC v1.8.7. See https://fcp-indi.github.io/docs/nightly/user/anat#configuring-fsl-bet-options for more information. @ data['anatomical_preproc']['brain_extraction']['FSL-BET']['mask_boolean']
   ```

   instead of one like

   ```
   MultipleInvalid: extra keys not allowed @ data['anatomical_preproc']['brain_extraction']['FSL-BET']['mask_boolean']
   ```

(`Invalid` can just be added to the `voluptuous` imports https://github.com/FCP-INDI/C-PAC/blob/cda064cf7b0bfbe8b837f75eca6d8572e6f6c6be/CPAC/pipeline/schema.py#L24-L27 and `DOCS_URL_PREFIX` is replacing `docs_prefix` in #2056)

---

- [ ] I think this change is significant enough to merit a line in the CHANGELOG (I'd add a `## Removed` header in https://github.com/FCP-INDI/C-PAC/blob/cda064cf7b0bfbe8b837f75eca6d8572e6f6c6be/CHANGELOG.md?plain=1#L17-L21)

---

- [x] As far as I can tell, `randomise` takes `mask_boolean` as an input https://github.com/FCP-INDI/C-PAC/blob/cda064cf7b0bfbe8b837f75eca6d8572e6f6c6be/CPAC/randomise/pipeline.py#L54 but never does anything with it, so I think we can remove if from that inputspec and https://github.com/FCP-INDI/C-PAC/blob/cda064cf7b0bfbe8b837f75eca6d8572e6f6c6be/CPAC/pipeline/cpac_randomise_pipeline.py#L66 as well

---

<span id="ci-failure">

- [This CI failure](https://github.com/FCP-INDI/C-PAC/actions/runs/7658939357/job/20873059221#step:10:160)

   ```Dockerfile
   Error: buildx failed with: ERROR: failed to solve: ghcr.io/fcp-indi/c-pac/stage-base:lite-v1.8.6: ghcr.io/fcp-indi/c-pac/stage-base:lite-v1.8.6: not found
   ```

   is from a bug introduced in #2046; I'll write up an issue (#2055) and open a PR (#2056) to fix that. It's unrelated to this PR.</span>

**Comment by sgiavasis** [comment]
Re: @shnizzedy - we can remove it from randomise yes.

Also reminder @birajstha to please update the changelog.

File: CPAC/resources/configs/pipeline_config_benchmark-ANTS.yml [hunk 1]
```diff
@@ -73,13 +73,6 @@ anatomical_preproc:
   acpc_alignment:
     T1w_brain_ACPC_template: $FSLDIR/data/standard/MNI152_T1_1mm_brain.nii.gz
```
Comment (shnizzedy) [inline CPAC/resources/configs/pipeline_config_benchmark-ANTS.yml:75]: I think we want to keep

```suggestion

  brain_extraction:
    run: On
```

here?

**Comment by shnizzedy** [comment]
Oh, I also think you can do like

```shell
git fetch origin
git checkout origin/develop -- .github/Dockerfiles/*.Dockerfile
```

and add and commit those changes to fix the merge conflicts. They're just parallel version comment updates

File: CHANGELOG.md [hunk 2]
```diff
@@ -195,6 +195,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Updated CI to only rebuild software dependencies on change
 - Replaced deprecated `optparse.OptionError` with to `click.BadParameter`
 - Relicensed C-PAC from BSD-3-Clause to LGPL-3.0-or-later
+- Updated `FSL-BET` config to default `-mask-boolean` flag as on, and removed all removed `mask-boolean` keys from configs.
```
Comment (shnizzedy) [inline CHANGELOG.md:198]: This should go under ["unreleased"](https://github.com/FCP-INDI/C-PAC/blob/fee3e532e773d94811b73dad3964591e6d3d3f56/CHANGELOG.md#unreleased), not under ["1.8.4"](https://github.com/FCP-INDI/C-PAC/blob/fee3e532e773d94811b73dad3964591e6d3d3f56/CHANGELOG.md#changed-3)

Comment (birajstha) [reply CHANGELOG.md:198]: Ooops sorry, didn't paid much attention there. Thanks for catching this ! Realy appreciate your feedback

Comment (shnizzedy) [reply CHANGELOG.md:199]:
```suggestion
```

Thanks for adding it to `Unreleased`. We still need to remove it from `1.8.4` though

File: CPAC/resources/configs/pipeline_config_abcd-prep.yml [hunk 3]
```diff
@@ -46,6 +46,7 @@ anatomical_preproc:
     T1w_ACPC_template: /opt/dcan-tools/pipeline/global/templates/MNI152_T1_1mm.nii.gz
 
   brain_extraction:
+    run: On
```
Comment (shnizzedy) [inline CPAC/resources/configs/pipeline_config_abcd-prep.yml:49]: Is this change intentional?

File: CPAC/resources/configs/pipeline_config_regtest-1.yml [hunk 4]
```diff
@@ -75,10 +75,10 @@ anatomical_preproc:
 
   brain_extraction:
     run: On
-    FSL-BET:
 
-      # Mask created along with skull stripping. It should be `On`, if selected functionalMasking :  ['Anatomical_Refined'] and `FSL` as skull-stripping method.
-      mask_boolean: Off
+    # using: ['3dSkullStrip', 'BET', 'UNet', 'niworkflows-ants', 'FreeSurfer-ABCD', 'FreeSurfer-BET-Tight', 'FreeSurfer-BET-Loose', 'FreeSurfer-Brainmask']
+    # this is a fork option
+    using: [BET]
```
Comment (shnizzedy) [inline CPAC/resources/configs/pipeline_config_regtest-1.yml:81]: Is this change intentional?

Comment (birajstha) [reply CPAC/resources/configs/pipeline_config_regtest-1.yml:81]: Yes, if I remove `mask_boolean` key and leave empty `FSL-BET` then there would be an error. Should I remove the `using : [BET]` line to let it choose default method instead?

File: CPAC/resources/configs/pipeline_config_abcd-prep.yml (hunk 3 above)
Comment (birajstha) [reply CPAC/resources/configs/pipeline_config_abcd-prep.yml:49]: Yes, but I think I should leave this untouched. This config inherits from `blank` and `brain extraction` is `off` there by default. I will revert it back.

File: CPAC/resources/configs/pipeline_config_regtest-1.yml (hunk 4 above)
Comment (birajstha) [reply CPAC/resources/configs/pipeline_config_regtest-1.yml:81]: Thanks Jon, Its clear. What I wanted was to remove `mask_boolean` without affecting the original workflow. 
You are right .. The `FSL-BET` key was there but Afni tool was used.
I will revert the changes back.
Thanks again for details

File: CPAC/resources/configs/pipeline_config_abcd-prep.yml (hunk 3 above)
Comment (shnizzedy) [reply CPAC/resources/configs/pipeline_config_abcd-prep.yml]: Resolved in 58622451dcc56c1d3e74bbf98f46bfc74dc40da6

File: CPAC/resources/configs/pipeline_config_regtest-1.yml (hunk 4 above)
Comment (shnizzedy) [reply CPAC/resources/configs/pipeline_config_regtest-1.yml]: Resolved in 58622451dcc56c1d3e74bbf98f46bfc74dc40da6

**Review by shnizzedy — APPROVED** [review]
Looks good to me! Thanks for the updates!

**Review by sgiavasis — APPROVED** [review]
